app/src/main/java/edu/ucsc/cmps115_spring2017/face2name/MainScreen.java

```diff
@@ -67,42 +63,73 @@ public void onPreviewReady() {
         int touchX = (int) event.getX();
         int touchY = (int) event.getY();
-        Log.e("PRESS", "GET_X: "+touchX+ " GET_Y: "+touchY);
-        Log.e("PRESS", "EVENT ACTION: "+event.getAction()+ " ACTION BUTTON PRESS: "+ MotionEvent.ACTION_BUTTON_PRESS);
-        if (event.getAction() == MotionEvent.ACTION_BUTTON_PRESS ||
-                event.getAction() == MotionEvent.ACTION_BUTTON_RELEASE ||
-                event.getAction() == MotionEvent.ACTION_DOWN ||
-                event.getAction() == MotionEvent.ACTION_UP) {
-            Log.e("PRESS", "RECT PRESSED");
-            if ((touchX >= l_x && touchX <= r_x) && (touchY >= u_y && touchY <= l_y)) {
-                AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(
-                        this);
-
-                // set title
-                alertDialogBuilder.setTitle("Rectangle Pressed");
-
-                // create alert dialog
-                AlertDialog alertDialog = alertDialogBuilder.create();
-
-                // show it
-                alertDialog.show();
+
+
+        if (isPressEvent(event.getAction()))
+        {
+            // try/catch block because getRectPoint throws an IllegalArgumentException
+            try{
+                //returns the coordinates of the red rectangle
+                int leftXCoordinate = mLayerView.getRectPoint(LayerView.coordinate_sections.LEFT_X);
+                int rightXCoordinate = mLayerView.getRectPoint(LayerView.coordinate_sections.RIGHT_X);
+                int upperYCoordinate = mLayerView.getRectPoint(LayerView.coordinate_sections.UPPER_Y);
+                int lowerYCoordinate = mLayerView.getRectPoint(LayerView.coordinate_sections.LOWER_Y);
+
+                //If the press was within the X bounds and within the Y bounds, continue
+                if ((touchX >= leftXCoordinate && touchX <= rightXCoordinate) &&
+                        (touchY >= upperYCoordinate && touchY <= lowerYCoordinate)) {
+
+                   //build dialong
+                    AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
+
+                    // set title
+                    alertDialogBuilder.setTitle("Rectangle Pressed");
+
+                    // create alert dialog
+                    AlertDialog alertDialog = alertDialogBuilder.create();
+
+                    // show it
+                    alertDialog.show();
+                }
+            }
+            catch(IllegalArgumentException iArg){
+                    Log.d("ILLEGALARGUMENT", "IllegalArgumentException: " + iArg);
             }
         }
         return true;
     }
 
     private CameraPreview mCameraPreview;
     private LayerView mLayerView;
+
+    //Custom constants as a replacement for MotionEvent.constants
+    private static final int ACTION_DOWN = 0;
+    private static final int ACTION_UP = 1;
+    private static final int ACTION_BUTTON_PRESS = 11;
+    private static final int ACTION_BUTTON_RELEASE = 12;
```

Comment:
You missed my point about `MotionEvent.ACTION_BUTTON_PRESS`. This constant does not exist before API version 23, so it's not possible for `event.getAction()` to return it in versions before that. Defining your own version of `ACTION_BUTTON_PRESS` won't do anything. And besides, it's a maintenance nightmare. What if the value of `MotionEvent.ACTION_BUTTON_PRESS` changes? You have to find another method of determining if a button has been pressed, one that is supported by API version 16 and up.